### PR TITLE
[ready for core review]Code Cleanup, Removed cruft, generalized code

### PR DESCRIPTION
**Pull request by pradpnayak**

### CODE REVIEW

**Comment by pradpnayak** [comment]
Could one of the admins please verify this patch?

**Comment by pradpnayak** [comment]
Jenkin test this please

**Review by monishdeb — APPROVED** [review]
Tested and the overall changes look good to me. Nice work :)

**Comment by monishdeb** [comment]
@eileenmcnaughton good to merge ?

**Comment by eileenmcnaughton** [comment]
I took a look and agree it is an improvement and everything I checked seemed to  be correct. I think test coverage is fairly good here. I have just one issue which is the addition of the isQuick to the getLineItemsByContributionID function. In general we are trying to get rid of the logic around that setting as it was only ever intended to be a display setting. I think it's possible to not pass that through from the 4 places you have added it on conversion but checking...

File: CRM/Contribute/BAO/Contribution.php
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: I tested removing the 'TRUE' on this line locally and then generating message receipts through the 'print contribution receipts' action off advanced search. My conclusion - it is currently broken & not displaying membership line items.

This commit https://github.com/civicrm/civicrm-core/pull/9327 is my fix for it - using your principle but moving the quick config concept up to the message template and letting the template decide whether to show the line items.

File: CRM/Contribute/Form/Contribution.php
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution.php]: We should try to eliminate the 'TRUE' hear or leave it unchanged for now & add a code comment explaining

File: CRM/Contribute/Form/ContributionView.php
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/ContributionView.php]: I propose that we can avoid passing the 'true with mysterious implications' by this commit https://github.com/civicrm/civicrm-core/pull/9327/commits/59f7b38eb943851563395eb57c386aa19c239443

File: CRM/Financial/BAO/FinancialAccount.php
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: We should either remove the TRUE or comment this & leave the change to next time

File: CRM/Price/BAO/LineItem.php
Comment (eileenmcnaughton) [inline CRM/Price/BAO/LineItem.php]: I really don't want to see isQuick added to this signature

File: CRM/Contribute/BAO/Contribution.php
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: Looks like my attempt above broke the tests. I have a suspicion it is either a leakage problem or the tests might need fixing. Regardless the way to not get blocked is to hold off on changing this line & comment that it should be updated to not require the isQuick - that way at least this PR can be merged

**Review by eileenmcnaughton — COMMENTED** [review]
It's good overall but I am very reluctant to see $isQuick added to that function signature. I have an alternative for that on the View form and thought I did on the receipt but it is not quite there.

My suggestion is to not change those 4 places yet & comment them instead. There are a bunch of good changes in here

File: CRM/Financial/BAO/FinancialAccount.php
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: Actually you are probably the best person to review this function & see if we can avoid referring to $isQuick from it

**Comment by eileenmcnaughton** [comment]
@pradpnayak I'm closing this as it is stale after the part I changed got tested & merged. However, I have put most of it into #9338 which I am able to merge quickly as they are changes of yours that I have reviewed. There are a few changes not brought across - ie. those relating to the $isQuick var as discussed above

**Pull request closed without merging**
